viahtml/hooks/hooks.py

```diff
@@ -27,14 +36,21 @@ def __init__(self, config):
         self.config = config
         self.context = None
 
-    def set_context(self, context):
-        """Set the request context for access in hook points."""
+    def set_context(self, context: Context):
+        """Set the current request context.
+
+        This must be called before the `modify_*` hooks are invoked.
```

Comment:
There were some implicit assumptions in existing `Hooks` methods about whether `context` is available. I notes and asserts to clarify where context is/is not expected to be available.

Reply:
Hooks are a bit strange, they try to abstract the patching over `patch.py` but I don't think it does a great job you still need to understand booth + our abstraction. 

These assets and docs will help avoiding mistakes for now :+1:

Reply:
> Hooks are a bit strange, they try to abstract the patching over patch.py but I don't think it does a great job you still need to understand booth + our abstraction.

I think the idea is that `patch.py` handles the mechanism of patching, whereas `hooks.py` handles the policy (ie. what does the patching do). That's reasonable, although the hooks still end up somewhat coupled to how pywb works.